pkg/accountmanager/poller.go

```diff
@@ -147,42 +146,51 @@ func (p *accountPoller) doAccountPolling() {
 
 	p.PollDone = false
 	// Ignoring error since it is captured in the CloudProviderAccount CR's status field.
-	_ = p.cloudInterface.DoInventoryPoll(p.namespacedName)
+	_ = p.cloudInterface.DoInventoryPoll(p.accountNamespacedName)
 
 	defer func() {
 		p.PollDone = true
 		// Update status on CPA CR after polling.
 		p.updateAccountStatus(p.cloudInterface)
 	}()
 
-	// TODO: Avoid calling plugin to get VPC inventory from snapshot.
-	vpcMap, err := p.cloudInterface.GetVpcInventory(p.namespacedName)
+	cloudInventory, err := p.cloudInterface.GetCloudInventory(p.accountNamespacedName)
 	if err != nil {
-		p.log.Error(err, "failed to fetch cloud vpc list from internal snapshot", "account",
-			p.namespacedName.String())
+		p.log.Error(err, "failed to fetch cloud inventory from internal snapshot", "account",
+			p.accountNamespacedName)
 		return
 	}
-	_ = p.inventory.BuildVpcCache(vpcMap, p.namespacedName)
-
-	// Perform VM Operations only when CES is added.
-	vmCount := 0
-	if p.selector != nil {
-		// TODO: Avoid calling plugin to get VM inventory from snapshot.
-		virtualMachines := p.getComputeResources(p.cloudInterface)
-		// TODO: We are walking thru virtual map twice. Once here and second one in BuildVmCAche.
-		// May be expose Add, Delete, Update routine in inventory and we do the calculation here.
-		p.updateAgentState(virtualMachines)
-		p.inventory.BuildVmCache(virtualMachines, p.namespacedName)
-		vmCount = len(virtualMachines)
+
+	p.buildVpcInventoryFromSnapshot(cloudInventory.VpcMap)
```

Comment:
Call it processCloudInventory and add vpc and vm handling there..

Reply:
done